src/iso15118/io/connection_ssl.cpp

```diff
@@ -1,288 +1,370 @@
         const auto msg =
-            "Failed to determine string representation of ipv6 socket address for interface " + interface_name;
+            "Failed to determine string representation of ipv6 socket address for interface " + interface_name_;
         log_and_throw(msg.c_str());
     }
 
+    // Todo(sl): Define constexpr -> 50000 is fixed?
     end_point.port = 50000;
     memcpy(&end_point.address, &address.sin6_addr, sizeof(address.sin6_addr));
 
-    //
-    // mbedtls specifica
-    //
+    // Openssl stuff missing!
+    init_ssl(ssl_config);
 
-    // initialize pseudo random number generator
-    // FIXME (aw): what is the custom argument about?
-    const auto ctr_drbg_seed_result =
-        mbedtls_ctr_drbg_seed(&ssl->ctr_drbg, mbedtls_entropy_func, &ssl->entropy, nullptr, 0);
-    if (ctr_drbg_seed_result != 0) {
-        log_and_raise_mbed_error("Failed in mbedtls_ctr_drbg_seed()", ctr_drbg_seed_result);
+    fd = socket(AF_INET6, SOCK_STREAM, 0);
+    if (fd == -1) {
+        log_and_throw("Failed to create an ipv6 socket");
     }
 
-    // load certificate
-    load_certificates(*ssl, ssl_config);
+    // before bind, set the port
+    address.sin6_port = htobe16(end_point.port);
 
-    const auto bind_result = mbedtls_net_bind(&ssl->accepting_net_ctx, address_name.get(),
-                                              std::to_string(end_point.port).c_str(), MBEDTLS_NET_PROTO_TCP);
-    if (bind_result != 0) {
-        log_and_raise_mbed_error("Failed to mbedtls_net_bind()", bind_result);
+    const auto bind_result = bind(fd, reinterpret_cast<const struct sockaddr*>(&address), sizeof(address));
+    if (bind_result == -1) {
+        const auto error = "Failed to bind ipv6 socket to interface " + interface_name_;
+        log_and_throw(error.c_str());
     }
 
-    // setup ssl context configuration
-    const auto ssl_config_result = mbedtls_ssl_config_defaults(
-        &ssl->conf, MBEDTLS_SSL_IS_SERVER, MBEDTLS_SSL_TRANSPORT_STREAM, MBEDTLS_SSL_PRESET_DEFAULT);
-    if (ssl_config_result != 0) {
-        log_and_raise_mbed_error("Failed to mbedtls_ssl_config_defaults()", ssl_config_result);
+    const auto listen_result = listen(fd, DEFAULT_SOCKET_BACKLOG);
+    if (listen_result == -1) {
+        log_and_throw("Listen on socket failed");
     }
 
-    mbedtls_ssl_conf_rng(&ssl->conf, mbedtls_ctr_drbg_random, &ssl->ctr_drbg);
+    poll_manager.register_fd(fd, [this]() { this->handle_connect(); });
+}
 
-    mbedtls_ssl_conf_dbg(
-        &ssl->conf,
-        [](void* callback_context, int debug_level, const char* file_name, int line_number, const char* message) {
-            logf("mbedtls debug (level: %d) - %s\n", debug_level, message);
-        },
-        stdout);
+ConnectionSSL::~ConnectionSSL() = default;
 
-    if (ssl_config.enable_ssl_logging) {
-        mbedtls_debug_set_threshold(3);
+void ConnectionSSL::set_event_callback(const ConnectionEventCallback& callback) {
+    this->event_callback = callback;
```

Comment:
I feel like we should remove the `this->` part here for consistency

Reply:
Good point!